airflow/jobs.py

```diff
@@ -2039,6 +2039,14 @@ def __init__(
 
     def _execute(self):
         self.task_runner = get_task_runner(self)
+
+        def signal_handler(signum, frame):
+            '''Setting kill signal handler'''
+            logging.error("Killing subprocess")
```

Comment:
SIGTERM is a pretty normal signal. logging.debug is probably sufficient (with pid or so included)